pkg/reconciler/taskrun/taskrun.go

```diff
@@ -370,7 +369,10 @@ func (c *Reconciler) finishReconcileUpdateEmitEvents(ctx context.Context, tr *v1
 	// Send k8s events and cloud events (when configured)
 	events.Emit(ctx, beforeCondition, afterCondition, tr)
 
-	merr := multierror.Append(previousError).ErrorOrNil()
+	var errs []error
+	if previousError != nil {
+		errs = append(errs, previousError)
+	}
```

Comment:
Whether or not we opt to use a `errors.Join` instead of `append(errs, err)`, we can take advantage of errors.Join's ability to ignore nil input:
(if using `errors.Join` for each error - if `previousErr` is nil `errors.Join()` will ignore it at the next call)
```suggestion
	errs := previousError
```
(if using `append(errs, err)` for each error, `errors.Join` will ignore a `nil` entry in the list when we return)
```suggestion
	errs := []error{previousError}
```